// primeng-extensions/components/qrcode/qrcode.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var QRCode = /** @class */ (function () {
    function QRCode() {
        this.background = 'white';
        this.backgroundAlpha = 1.0;
        this.foreground = 'black';
        this.foregroundAlpha = 1.0;
        this.level = 'L';
        this.mime = 'image/png';
        this.padding = null;
        this.size = 100;
    }
    QRCode.prototype.ngOnInit = function () {
        this.initialized = false;
    };
    QRCode.prototype.ngAfterViewChecked = function () {
        if (!this.initialized) {
            this.init();
        }
    };
    QRCode.prototype.init = function () {
        this.initialized = true;
        var qrObject = {
            element: this.containerViewChild.nativeElement,
            background: this.background,
            backgroundAlpha: this.backgroundAlpha,
            foreground: this.foreground,
            foregroundAlpha: this.foregroundAlpha,
            level: this.level,
            mime: this.mime,
            padding: this.padding,
            size: this.size,
            value: this.value
        };
        this.qrcode = new QRious(qrObject);
    };
    QRCode.prototype.ngOnDestroy = function () {
        this.qrcode = null;
        this.initialized = false;
    };
    __decorate([
        core_1.Input(),
        __metadata("design:type", String)
    ], QRCode.prototype, "value", void 0);
    __decorate([
        core_1.Input(),
        __metadata("design:type", String)
    ], QRCode.prototype, "background", void 0);
    __decorate([
        core_1.Input(),
        __metadata("design:type", Number)
    ], QRCode.prototype, "backgroundAlpha", void 0);
    __decorate([
        core_1.Input(),
        __metadata("design:type", String)
    ], QRCode.prototype, "foreground", void 0);
    __decorate([
        core_1.Input(),
        __metadata("design:type", Number)
    ], QRCode.prototype, "foregroundAlpha", void 0);
    __decorate([
        core_1.Input(),
        __metadata("design:type", String)
    ], QRCode.prototype, "level", void 0);
    __decorate([
        core_1.Input(),
        __metadata("design:type", String)
    ], QRCode.prototype, "mime", void 0);
    __decorate([
        core_1.Input(),
        __metadata("design:type", Number)
    ], QRCode.prototype, "padding", void 0);
    __decorate([
        core_1.Input(),
        __metadata("design:type", Number)
    ], QRCode.prototype, "size", void 0);
    __decorate([
        core_1.ViewChild('qrcode'),
        __metadata("design:type", core_1.ElementRef)
    ], QRCode.prototype, "containerViewChild", void 0);
    QRCode = __decorate([
        core_1.Component({
            selector: 'pe-qrcode',
            template: "\n    <div class=\"ui-widget ui-corner-all\">\n      <canvas #qrcode></canvas>\n    </div>"
        })
    ], QRCode);
    return QRCode;
}());
exports.QRCode = QRCode;
var QRCodeModule = /** @class */ (function () {
    function QRCodeModule() {
    }
    QRCodeModule = __decorate([
        core_1.NgModule({
            imports: [common_1.CommonModule],
            exports: [QRCode],
            declarations: [QRCode]
        })
    ], QRCodeModule);
    return QRCodeModule;
}());
exports.QRCodeModule = QRCodeModule;
